management/api-client.go

```diff
@@ -0,0 +1,109 @@
+package ziti
+
+import (
+	"crypto/tls"
+	"github.com/go-openapi/runtime"
+	openapiclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/openziti/edge-api/rest_management_api_client"
+	"github.com/openziti/edge-api/rest_util"
+	"net"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"time"
+)
+
+type AuthenticatedApiClient struct {
+	*ApiClient
+	ZtSession     string
+	Authorization []string
+}
+
+func (a AuthenticatedApiClient) AuthenticateRequest(request runtime.ClientRequest, _ strfmt.Registry) error {
```

Comment:
this function is misnamed imo. It's more like "add authentiation headers" but also - does this need to be public?

Reply:
It is an interface implementation. I do not get the choice.